arangod/Aql/ExecutionBlockImpl.tpp

```diff
@@ -982,12 +900,99 @@ auto ExecutionBlockImpl<Executor>::executeSkipRowsRange(
   THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL);
 }
 
-// Avoid duplicate symbols when linking: This file is directly included by
-// ExecutionBlockImplTestInstances.cpp
-#ifndef ARANGODB_INCLUDED_FROM_GTESTS
-template<>
-auto ExecutionBlockImpl<SubqueryStartExecutor>::shadowRowForwarding(
-    AqlCallStack& stack) -> ExecState {
+template<class Executor>
+auto ExecutionBlockImpl<Executor>::sideEffectShadowRowForwarding(
+    AqlCallStack& stack, SkipResult& skipResult) -> ExecState {
+  TRI_ASSERT(executorHasSideEffects<Executor>);
```

Comment:
`executorHasSideEffects` should be constexpr - can we make this a static_assert?